tcping.go

```diff
@@ -377,6 +391,77 @@ func permuteArgs(args cliArgs) {
 	}
 }
 
+// newNetworkInterface uses the 1st ip address of the interface
+// if any err occurs it calls `tcpStats.printer.printError` and exits with statuscode 1.
+// or return `networkInterface`
+func newNetworkInterface(tcpStats *stats, netInterface string) networkInterface {
+	var interfaceAddress net.IP
+
+	// if netinterface is the addres `interfaceAddress` var will not be `nil`
+	interfaceAddress = net.ParseIP(netInterface)
+
+	if interfaceAddress == nil {
+		ief, err := net.InterfaceByName(netInterface)
+		if err != nil {
+			tcpStats.printer.printError("Interface %s not found", netInterface)
+			os.Exit(1)
+		}
+
+		addrs, err := ief.Addrs()
+		if err != nil {
+			tcpStats.printer.printError("Unable to get Interface addresses")
+			os.Exit(1)
+		}
+
+		// Iterating through the available addresses to identify valid IP configurations
+		for _, addr := range addrs {
+			if ip := addr.(*net.IPNet).IP; ip != nil {
+				// netip.Addr
+				nipAddr, err := netip.ParseAddr(ip.String())
+				if err != nil {
+					continue
+				}
+
+				if nipAddr.Is4() && !tcpStats.userInput.useIPv6 {
+					interfaceAddress = ip
+					break
+				} else if nipAddr.Is6() && !tcpStats.userInput.useIPv4 {
+					interfaceAddress = ip
+					break
```

Comment:
@wizsk here we need to check and `continue` if the IPv6 is a link-local address or not using this method https://pkg.go.dev/net/netip#Addr.IsLinkLocalUnicast